src/commands/deployImage/imageSource/containerRegistry/acr/createAcr/RegistryNameStep.ts

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { ContainerRegistryManagementClient, RegistryNameStatus } from "@azure/arm-containerregistry";
+import { AzureWizardPromptStep } from "@microsoft/vscode-azext-utils";
+import { createContainerRegistryManagementClient } from "../../../../../../utils/azureClients";
+import { localize } from "../../../../../../utils/localize";
+import { ICreateAcrContext } from "./ICreateAcrContext";
+
+export class RegistryNameStep extends AzureWizardPromptStep<ICreateAcrContext> {
+    public async prompt(context: ICreateAcrContext): Promise<void> {
+        context.registryName = await context.ui.showInputBox({
+            prompt: localize('registryName', 'Enter a name for the new registry'),
+            validateInput: async (value: string | undefined): Promise<string | undefined> => await this.validateInput(context, value)
+        });
+    }
+
+    public shouldPrompt(context: ICreateAcrContext): boolean {
+        return !context.registryName;
+    }
+
+    private async validateInput(context: ICreateAcrContext, name: string | undefined): Promise<string | undefined> {
+        name = name ? name.trim() : '';
+
+        const min = 5;
+        const max = 50;
+        if (name.length < min || name.length > max) {
+            return localize('validationLengthError', 'The name must be between {0} and {1} characters.', min, max);
+        } else if (!/^[a-z][a-zA-Z0-9]*$/.test(name)) {
+            return localize('validateInputError', `Connection names can only consist of alphanumeric characters.`); //make it between 5 and 50 character
+        } else {
+            const client: ContainerRegistryManagementClient = await createContainerRegistryManagementClient(context);
```

Comment:
So there's a separate property on `context.ui.showInputBox` now that we added called `asyncValidationTask` that only runs at the end when the user presses enter.  This is nice to use for validation requests like this because it condenses what would otherwise be a network call on each keystroke into potentially just one or a few at the end.  Hopefully there are some good examples already in the repo, if not feel free to reach out!